{"cells":[{"cell_type":"markdown","id":"INpkndIxRH2Q","metadata":{"id":"INpkndIxRH2Q"},"source":["# cat_col 상관관계 분석"]},{"cell_type":"code","execution_count":1,"id":"2a4LKhsK2_P2","metadata":{"executionInfo":{"elapsed":10,"status":"ok","timestamp":1668392173559,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"2a4LKhsK2_P2"},"outputs":[],"source":["import pandas as pd\n","import numpy as np\n","\n","train_df = pd.read_csv(r'C:\\A\\breastCancer\\ml/train.csv')\n","test_df = pd.read_csv(r\"C:\\A\\breastCancer\\ml/test.csv\")\n"]},{"cell_type":"code","execution_count":2,"id":"09ENKEWrLWOD","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":3,"status":"ok","timestamp":1668392173870,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"09ENKEWrLWOD","outputId":"f837083b-41b2-47e7-e27d-4562aaab806d"},"outputs":[],"source":["import os\n","try:\n","  import google.colab\n","  IN_COLAB = True\n","except:\n","  IN_COLAB = False\n","if not IN_COLAB:\n","    os.chdir(\"..\")\n","from sklearn.datasets import fetch_covtype\n","import random\n","import numpy as np\n","import pandas as pd\n","import lightgbm as lgb\n","from sklearn.ensemble import RandomForestClassifier\n","from sklearn.metrics import accuracy_score, f1_score\n","from sklearn.model_selection import train_test_split\n","%load_ext autoreload\n","%autoreload 2\n"]},{"cell_type":"code","execution_count":3,"id":"j6QteYPJ3-tb","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":3905,"status":"ok","timestamp":1668392178200,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"j6QteYPJ3-tb","outputId":"6ceb22f8-6cf1-4214-e1a2-069309170bee"},"outputs":[],"source":["import numpy as np # linear algebra\n","import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\n","import matplotlib.pyplot as plt # data visualization\n","import seaborn as sns # statistical data visualization\n","import os\n","import warnings\n","warnings.filterwarnings(\"ignore\")\n"]},{"cell_type":"code","execution_count":4,"id":"qjCvwUQWep-w","metadata":{"executionInfo":{"elapsed":7,"status":"ok","timestamp":1668392178201,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"qjCvwUQWep-w"},"outputs":[{"name":"stderr","output_type":"stream","text":["[autoreload of sklearn.base failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: clone() requires a code object with 0 free vars, not 3\n","]\n","[autoreload of sklearn.utils failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\sklearn\\utils\\__init__.py\", line 27, in <module>\n","    from .fixes import np_version, parse_version\n","ImportError: cannot import name 'parse_version' from 'sklearn.utils.fixes' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\sklearn\\utils\\fixes.py)\n","]\n","[autoreload of scipy.sparse.base failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\base.py\", line 4, in <module>\n","    from .sputils import (isdense, isscalarlike, isintlike,\n","ImportError: cannot import name 'asmatrix' from 'scipy.sparse.sputils' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\sputils.py)\n","]\n","[autoreload of scipy.sparse.sputils failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\sputils.py\", line 8, in <module>\n","    from scipy._lib._util import prod\n","ImportError: cannot import name 'prod' from 'scipy._lib._util' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\_lib\\_util.py)\n","]\n","[autoreload of scipy.sparse.compressed failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\compressed.py\", line 14, in <module>\n","    from ._sparsetools import (get_csr_submatrix, csr_sample_offsets, csr_todense,\n","ImportError: cannot import name 'csr_row_index' from 'scipy.sparse._sparsetools' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\_sparsetools.cp37-win_amd64.pyd)\n","]\n","[autoreload of scipy.sparse.data failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\data.py\", line 12, in <module>\n","    from .sputils import isscalarlike, validateaxis, matrix\n","ImportError: cannot import name 'matrix' from 'scipy.sparse.sputils' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\sputils.py)\n","]\n","[autoreload of scipy.sparse.dia failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\dia.py\", line 11, in <module>\n","    from .sputils import (isshape, upcast_char, getdtype, get_index_dtype,\n","ImportError: cannot import name 'matrix' from 'scipy.sparse.sputils' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\sputils.py)\n","]\n","[autoreload of scipy.sparse.lil failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\lil.py\", line 14, in <module>\n","    from .sputils import (getdtype, isshape, isscalarlike, upcast_scalar,\n","ImportError: cannot import name 'asmatrix' from 'scipy.sparse.sputils' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\sputils.py)\n","]\n","[autoreload of scipy.sparse.coo failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\coo.py\", line 15, in <module>\n","    from .sputils import (upcast, upcast_char, to_native, isshape, getdtype,\n","ImportError: cannot import name 'getdata' from 'scipy.sparse.sputils' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\sputils.py)\n","]\n","[autoreload of scipy.sparse.bsr failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\bsr.py\", line 14, in <module>\n","    from .sputils import (isshape, getdtype, getdata, to_native, upcast,\n","ImportError: cannot import name 'getdata' from 'scipy.sparse.sputils' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\sputils.py)\n","]\n","[autoreload of sklearn.utils.class_weight failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: compute_class_weight() requires a code object with 0 free vars, not 3\n","]\n","[autoreload of sklearn.utils.fixes failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\sklearn\\utils\\fixes.py\", line 108, in <module>\n","    class loguniform(scipy.stats.reciprocal):\n","AttributeError: module 'scipy' has no attribute 'stats'\n","]\n","[autoreload of scipy.stats failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\stats\\__init__.py\", line 452, in <module>\n","    from ._bootstrap import bootstrap\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\stats\\_bootstrap.py\", line 7, in <module>\n","    from scipy.stats.stats import _broadcast_concatenate\n","ImportError: cannot import name '_broadcast_concatenate' from 'scipy.stats.stats' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\stats\\stats.py)\n","]\n","[autoreload of scipy.stats.stats failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\stats\\stats.py\", line 47, in <module>\n","    from ._stats import (_kendall_dis, _toint64, _weightedrankedtau,\n","ImportError: cannot import name '_local_correlations' from 'scipy.stats._stats' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\stats\\_stats.cp37-win_amd64.pyd)\n","]\n","[autoreload of scipy.special failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\special\\__init__.py\", line 649, in <module>\n","    from ._logsumexp import logsumexp, softmax, log_softmax\n","ImportError: cannot import name 'log_softmax' from 'scipy.special._logsumexp' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\special\\_logsumexp.py)\n","]\n","[autoreload of scipy.linalg.blas failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: get_blas_funcs() requires a code object with 0 free vars, not 2\n","]\n","[autoreload of scipy.linalg.lapack failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\linalg\\lapack.py\", line 885, in <module>\n","    ssyevx, dsyevx, cheevx, zheevx,\n","NameError: name 'ssyevx' is not defined\n","]\n","[autoreload of scipy.misc.doccer failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: docformat() requires a code object with 0 free vars, not 2\n","]\n","[autoreload of scipy.interpolate._fitpack_impl failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\interpolate\\_fitpack_impl.py\", line 38, in <module>\n","    dfitpack_int = dfitpack.types.intvar.dtype\n","AttributeError: module 'scipy.interpolate.dfitpack' has no attribute 'types'\n","]\n","[autoreload of scipy.interpolate.fitpack2 failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\interpolate\\fitpack2.py\", line 31, in <module>\n","    dfitpack_int = dfitpack.types.intvar.dtype\n","AttributeError: module 'scipy.interpolate.dfitpack' has no attribute 'types'\n","]\n","[autoreload of scipy.spatial.kdtree failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 1\n","]\n","[autoreload of scipy.spatial.distance failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: _validate_wminkowski_kwargs() requires a code object with 0 free vars, not 3\n","]\n","[autoreload of scipy.sparse.linalg.isolve.iterative failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\linalg\\isolve\\iterative.py\", line 150, in <module>\n","    def bicg(A, b, x0=None, tol=1e-5, maxiter=None, M=None, callback=None, atol=None):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\_lib\\_threadsafety.py\", line 59, in decorator\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\_lib\\_threadsafety.py\", line 47, in decorate\n","AttributeError: module 'scipy' has no attribute '_lib'\n","]\n","[autoreload of scipy.sparse.linalg.interface failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\linalg\\interface.py\", line 48, in <module>\n","    from scipy.sparse.sputils import isshape, isintlike, asmatrix, is_pydata_spmatrix\n","ImportError: cannot import name 'asmatrix' from 'scipy.sparse.sputils' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\sputils.py)\n","]\n","[autoreload of scipy.sparse.linalg.isolve.utils failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\linalg\\isolve\\utils.py\", line 7, in <module>\n","    from scipy.sparse.sputils import asmatrix\n","ImportError: cannot import name 'asmatrix' from 'scipy.sparse.sputils' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\sputils.py)\n","]\n","[autoreload of scipy.sparse.linalg.dsolve.linsolve failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\linalg\\dsolve\\linsolve.py\", line 7, in <module>\n","    from scipy.sparse.sputils import is_pydata_spmatrix\n","ImportError: cannot import name 'is_pydata_spmatrix' from 'scipy.sparse.sputils' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\sputils.py)\n","]\n","[autoreload of scipy.sparse.linalg.eigen.arpack.arpack failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\linalg\\eigen\\arpack\\arpack.py\", line 50, in <module>\n","    from scipy.sparse.sputils import isdense, is_pydata_spmatrix\n","ImportError: cannot import name 'is_pydata_spmatrix' from 'scipy.sparse.sputils' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\sputils.py)\n","]\n","[autoreload of scipy.sparse.linalg.matfuncs failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\linalg\\matfuncs.py\", line 21, in <module>\n","    from scipy.sparse.sputils import is_pydata_spmatrix\n","ImportError: cannot import name 'is_pydata_spmatrix' from 'scipy.sparse.sputils' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\sputils.py)\n","]\n","[autoreload of scipy.sparse.linalg._expm_multiply failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\linalg\\_expm_multiply.py\", line 9, in <module>\n","    from scipy.sparse.sputils import is_pydata_spmatrix\n","ImportError: cannot import name 'is_pydata_spmatrix' from 'scipy.sparse.sputils' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\sparse\\sputils.py)\n","]\n","[autoreload of scipy.optimize.cobyla failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: fmin_cobyla() requires a code object with 0 free vars, not 1\n","]\n","[autoreload of scipy.optimize._linprog failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\optimize\\_linprog.py\", line 29, in <module>\n","    from ._linprog_util import (\n","ImportError: cannot import name '_LPProblem' from 'scipy.optimize._linprog_util' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\optimize\\_linprog_util.py)\n","]\n","[autoreload of scipy.optimize._linprog_util failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\optimize\\_linprog_util.py\", line 9, in <module>\n","    from scipy.optimize._remove_redundancy import (\n","ImportError: cannot import name '_remove_redundancy_svd' from 'scipy.optimize._remove_redundancy' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\optimize\\_remove_redundancy.py)\n","]\n","[autoreload of scipy.integrate failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\integrate\\__init__.py\", line 95, in <module>\n","    from ._ivp import (solve_ivp, OdeSolution, DenseOutput,\n","ImportError: cannot import name 'DOP853' from 'scipy.integrate._ivp' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\integrate\\_ivp\\__init__.py)\n","]\n","[autoreload of scipy.integrate._ode failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\integrate\\_ode.py\", line 95, in <module>\n","    _dop_int_dtype = _dop.types.intvar.dtype\n","AttributeError: module 'scipy.integrate._dop' has no attribute 'types'\n","]\n","[autoreload of scipy.integrate._ivp failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\integrate\\_ivp\\__init__.py\", line 3, in <module>\n","    from .rk import RK23, RK45, DOP853\n","ImportError: cannot import name 'DOP853' from 'scipy.integrate._ivp.rk' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\integrate\\_ivp\\rk.py)\n","]\n","[autoreload of scipy.integrate._ivp.ivp failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\integrate\\_ivp\\ivp.py\", line 5, in <module>\n","    from .rk import RK23, RK45, DOP853\n","ImportError: cannot import name 'DOP853' from 'scipy.integrate._ivp.rk' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\integrate\\_ivp\\rk.py)\n","]\n","[autoreload of scipy.stats._continuous_distns failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: fit() requires a code object with 0 free vars, not 1\n","]\n","[autoreload of scipy.stats._stats_mstats_common failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: _asdict() requires a code object with 0 free vars, not 2\n","]\n","[autoreload of scipy.stats._rvs_sampling failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\stats\\_rvs_sampling.py\", line 4, in <module>\n","    from scipy.interpolate import CubicHermiteSpline\n","ImportError: cannot import name 'CubicHermiteSpline' from 'scipy.interpolate' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\interpolate\\__init__.py)\n","]\n","[autoreload of scipy.stats.morestats failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\stats\\morestats.py\", line 16, in <module>\n","    from .stats import find_repeats, _contains_nan, _normtest_finish\n","ImportError: cannot import name '_normtest_finish' from 'scipy.stats.stats' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\stats\\stats.py)\n","]\n","[autoreload of scipy.stats.kde failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\stats\\kde.py\", line 35, in <module>\n","    from ._stats import gaussian_kernel_estimate\n","ImportError: cannot import name 'gaussian_kernel_estimate' from 'scipy.stats._stats' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\stats\\_stats.cp37-win_amd64.pyd)\n","]\n","[autoreload of sklearn.utils.validation failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: assert_all_finite() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.datasets._base failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: load_files() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.datasets._covtype failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: fetch_covtype() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.datasets._kddcup99 failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: fetch_kddcup99() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.datasets._lfw failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: fetch_lfw_people() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.datasets._twenty_newsgroups failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: fetch_20newsgroups() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.feature_extraction._dict_vectorizer failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.feature_extraction.image failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: img_to_graph() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.feature_extraction.text failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.preprocessing._function_transformer failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.preprocessing._data failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: scale() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.utils.extmath failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: safe_sparse_dot() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.utils.sparsefuncs failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: incr_mean_variance_axis() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.preprocessing._encoders failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.preprocessing._label failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.preprocessing._discretization failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.datasets._openml failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: fetch_openml() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.datasets._samples_generator failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: make_classification() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.datasets._olivetti_faces failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: fetch_olivetti_faces() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of scipy.io failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\io\\__init__.py\", line 103, in <module>\n","    from ._fortran import FortranFile, FortranEOFError, FortranFormattingError\n","ImportError: cannot import name 'FortranEOFError' from 'scipy.io._fortran' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\scipy\\io\\_fortran.py)\n","]\n","[autoreload of sklearn.datasets._species_distributions failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: fetch_species_distributions() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.datasets._california_housing failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: fetch_california_housing() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.datasets._rcv1 failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: fetch_rcv1() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.model_selection._split failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.model_selection._validation failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: cross_validate() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.metrics._ranking failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: average_precision_score() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.metrics._classification failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: accuracy_score() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.metrics.cluster._supervised failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: contingency_matrix() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.metrics.cluster._unsupervised failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: silhouette_score() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.metrics.pairwise failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: check_pairwise_arrays() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.metrics.cluster._bicluster failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: consensus_score() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.metrics._regression failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: mean_absolute_error() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.metrics._scorer failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: check_scoring() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.metrics._plot.roc_curve failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: plot() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.metrics._plot.precision_recall_curve failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: plot() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.metrics._plot.confusion_matrix failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: plot() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.model_selection._search failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.ensemble._forest failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 1 free vars, not 4\n","]\n","[autoreload of sklearn.tree._classes failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.neighbors._graph failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: kneighbors_graph() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.neighbors._base failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\sklearn\\neighbors\\_base.py\", line 26, in <module>\n","    from ..utils import _to_object_array\n","ImportError: cannot import name '_to_object_array' from 'sklearn.utils' (c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\sklearn\\utils\\__init__.py)\n","]\n","[autoreload of sklearn.neighbors._unsupervised failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 1 free vars, not 4\n","]\n","[autoreload of sklearn.neighbors._classification failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 1 free vars, not 4\n","]\n","[autoreload of sklearn.neighbors._regression failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 1 free vars, not 4\n","]\n","[autoreload of sklearn.neighbors._nearest_centroid failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.neighbors._kde failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.neighbors._lof failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 1 free vars, not 4\n","]\n","[autoreload of sklearn.neighbors._nca failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.decomposition._nmf failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: non_negative_factorization() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.decomposition._pca failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.decomposition._incremental_pca failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.decomposition._kernel_pca failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.decomposition._sparse_pca failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.linear_model._base failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.linear_model._bayes failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.linear_model._least_angle failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: lars_path() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.linear_model._coordinate_descent failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: lasso_path() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.linear_model._huber failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.linear_model._stochastic_gradient failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.linear_model._ridge failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 394, in superreload\n","    module = reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\imp.py\", line 314, in reload\n","    return importlib.reload(module)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\importlib\\__init__.py\", line 169, in reload\n","    _bootstrap._exec(spec, module)\n","  File \"<frozen importlib._bootstrap>\", line 630, in _exec\n","  File \"<frozen importlib._bootstrap_external>\", line 728, in exec_module\n","  File \"<frozen importlib._bootstrap>\", line 219, in _call_with_frames_removed\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\sklearn\\linear_model\\_ridge.py\", line 987, in <module>\n","    class _X_CenterStackOp(sparse.linalg.LinearOperator):\n","AttributeError: module 'scipy.sparse' has no attribute 'linalg'\n","]\n","[autoreload of sklearn.linear_model._sag failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: sag_solver() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.linear_model._logistic failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.svm._classes failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.svm._bounds failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: l1_min_c() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.linear_model._omp failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: orthogonal_mp() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.linear_model._passive_aggressive failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 1 free vars, not 4\n","]\n","[autoreload of sklearn.linear_model._perceptron failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 1 free vars, not 4\n","]\n","[autoreload of sklearn.linear_model._ransac failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.linear_model._theil_sen failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.decomposition._dict_learning failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: sparse_encode() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.decomposition._truncated_svd failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.decomposition._fastica failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: fastica() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.decomposition._factor_analysis failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.tree._export failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: plot_tree() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.ensemble._bagging failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 1 free vars, not 4\n","]\n","[autoreload of sklearn.ensemble._iforest failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 1 free vars, not 4\n","]\n","[autoreload of sklearn.ensemble._weight_boosting failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 1 free vars, not 4\n","]\n","[autoreload of sklearn.ensemble._gb failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 1 free vars, not 4\n","]\n","[autoreload of sklearn.dummy failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 0 free vars, not 4\n","]\n","[autoreload of sklearn.ensemble._voting failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 1 free vars, not 4\n","]\n","[autoreload of sklearn.ensemble._stacking failed: Traceback (most recent call last):\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 245, in check\n","    superreload(m, reload, self.old_objects)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 410, in superreload\n","    update_generic(old_obj, new_obj)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 302, in update_class\n","    if update_generic(old_obj, new_obj): continue\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 347, in update_generic\n","    update(a, b)\n","  File \"c:\\Users\\wew12\\Anaconda3\\envs\\ed4\\lib\\site-packages\\IPython\\extensions\\autoreload.py\", line 266, in update_function\n","    setattr(old, name, getattr(new, name))\n","ValueError: __init__() requires a code object with 1 free vars, not 4\n","]\n"]}],"source":["# CAT_COL fillna\n","train_df['NG']= train_df['NG'].fillna(2)\n","train_df['HG']= train_df['HG'].fillna(2)\n","train_df['HG_score_1']= train_df['HG_score_1'].fillna(3)\n","train_df['HG_score_2']= train_df['HG_score_2'].fillna(2)\n","train_df['HG_score_3']= train_df['HG_score_3'].fillna(1)\n","train_df['DCIS_or_LCIS_type']= train_df['DCIS_or_LCIS_type'].fillna(2)\n","train_df['T_category']= train_df['T_category'].fillna(1)\n","train_df['ER']= train_df['ER'].fillna(1)\n","train_df['PR']= train_df['PR'].fillna(1)\n","train_df['HER2']= train_df['HER2'].fillna(0)\n","train_df['HER2_IHC']= train_df['HER2_IHC'].fillna(2)\n","train_df['HER2_SISH']= train_df['HER2_SISH'].fillna(0)\n","train_df['BRCA_mutation']= train_df['BRCA_mutation'].fillna(0)\n","\n","# NUL_COL fillna\n","# median\n","train_df['size of cancers'] = train_df['size of cancers'].fillna(train_df['size of cancers'].median())\n","train_df['ER_Allred_score'] = train_df['ER_Allred_score'].fillna(train_df['ER_Allred_score'].median())\n","train_df['PR_Allred_score'] = train_df['PR_Allred_score'].fillna(train_df['PR_Allred_score'].median())\n","train_df['KI-67_LI_percent'] = train_df['KI-67_LI_percent'].fillna(train_df['KI-67_LI_percent'].median())\n","train_df['HER2_SISH_ratio'] = train_df['HER2_SISH_ratio'].fillna(train_df['HER2_SISH_ratio'].median())\n","\n","\n","target_name = ['N_category']\n","\n","# 명목형 변수\n","cat_col_names = [\n","    'dx','location of cancers', 'number of cancers', 'NG','HG','HG_score_1','HG_score_2','HG_score_3','DCIS_or_LCIS','DCIS_or_LCIS_type','T_category','ER','PR','HER2','HER2_IHC','HER2_SISH',\"BRCA_mutation\"\n","]\n","\n","for col_name in cat_col_names:\n","    cat_values = np.unique(train_df[col_name])\n","    \n","    #print(f\"{col_name}: {cat_values}\\n\")\n","\n","\n","# 수치형 변수\n","num_col_names = [\n","    'age','size of cancers','ER_Allred_score','PR_Allred_score','KI-67_LI_percent','HER2_SISH_ratio'\n","]\n","\n","feature_columns = (\n","    num_col_names + cat_col_names + target_name)\n","\n","train_df = train_df.loc[:,feature_columns]\n","\n","\n","test = test_df.loc[:,num_col_names + cat_col_names]\n","\n","\n","X = train_df.loc[:,cat_col_names]\n","y = train_df.loc[:,target_name]\n"]},{"cell_type":"code","execution_count":5,"id":"3f4618be","metadata":{},"outputs":[{"name":"stdout","output_type":"stream","text":["<class 'pandas.core.frame.DataFrame'>\n","RangeIndex: 1000 entries, 0 to 999\n","Data columns (total 24 columns):\n"," #   Column             Non-Null Count  Dtype  \n","---  ------             --------------  -----  \n"," 0   age                1000 non-null   int64  \n"," 1   sizeofcancers      1000 non-null   float64\n"," 2   ER_Allred_score    1000 non-null   float64\n"," 3   PR_Allred_score    1000 non-null   float64\n"," 4   KI67_LI_percent    1000 non-null   float64\n"," 5   HER2_SISH_ratio    1000 non-null   float64\n"," 6   dx                 1000 non-null   int64  \n"," 7   locationofcancers  1000 non-null   int64  \n"," 8   numberofcancers    1000 non-null   int64  \n"," 9   NG                 1000 non-null   float64\n"," 10  HG                 1000 non-null   float64\n"," 11  HG_score_1         1000 non-null   float64\n"," 12  HG_score_2         1000 non-null   float64\n"," 13  HG_score_3         1000 non-null   float64\n"," 14  DCIS_or_LCIS       1000 non-null   int64  \n"," 15  DCIS_or_LCIS_type  1000 non-null   float64\n"," 16  T_category         1000 non-null   float64\n"," 17  ER                 1000 non-null   float64\n"," 18  PR                 1000 non-null   float64\n"," 19  HER2               1000 non-null   float64\n"," 20  HER2_IHC           1000 non-null   float64\n"," 21  HER2_SISH          1000 non-null   float64\n"," 22  BRCA_mutation      1000 non-null   float64\n"," 23  N_category         1000 non-null   int64  \n","dtypes: float64(18), int64(6)\n","memory usage: 187.6 KB\n"]}],"source":["import re\n","df = train_df.rename(columns = lambda x:re.sub('[^A-Za-z0-9_]+', '', x))\n","df.info()"]},{"cell_type":"code","execution_count":6,"id":"juZD_iV90MRX","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":6,"status":"ok","timestamp":1668392178201,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"juZD_iV90MRX","outputId":"62447721-f4f0-4b54-e79c-5e71dfbeb001"},"outputs":[{"name":"stdout","output_type":"stream","text":["<class 'pandas.core.frame.DataFrame'>\n","RangeIndex: 1000 entries, 0 to 999\n","Data columns (total 17 columns):\n"," #   Column               Non-Null Count  Dtype  \n","---  ------               --------------  -----  \n"," 0   dx                   1000 non-null   int64  \n"," 1   location of cancers  1000 non-null   int64  \n"," 2   number of cancers    1000 non-null   int64  \n"," 3   NG                   1000 non-null   float64\n"," 4   HG                   1000 non-null   float64\n"," 5   HG_score_1           1000 non-null   float64\n"," 6   HG_score_2           1000 non-null   float64\n"," 7   HG_score_3           1000 non-null   float64\n"," 8   DCIS_or_LCIS         1000 non-null   int64  \n"," 9   DCIS_or_LCIS_type    1000 non-null   float64\n"," 10  T_category           1000 non-null   float64\n"," 11  ER                   1000 non-null   float64\n"," 12  PR                   1000 non-null   float64\n"," 13  HER2                 1000 non-null   float64\n"," 14  HER2_IHC             1000 non-null   float64\n"," 15  HER2_SISH            1000 non-null   float64\n"," 16  BRCA_mutation        1000 non-null   float64\n","dtypes: float64(13), int64(4)\n","memory usage: 132.9 KB\n"]}],"source":["X.info()"]},{"cell_type":"code","execution_count":7,"id":"0kv_rh3V0QDO","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":5,"status":"ok","timestamp":1668392182406,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"0kv_rh3V0QDO","outputId":"c571a145-9ee7-446c-d5a6-8f0dd463be3c"},"outputs":[{"name":"stdout","output_type":"stream","text":["<class 'pandas.core.frame.DataFrame'>\n","RangeIndex: 1000 entries, 0 to 999\n","Data columns (total 1 columns):\n"," #   Column      Non-Null Count  Dtype\n","---  ------      --------------  -----\n"," 0   N_category  1000 non-null   int64\n","dtypes: int64(1)\n","memory usage: 7.9 KB\n"]}],"source":["y.info()"]},{"cell_type":"code","execution_count":8,"id":"7LN97U-b2LaD","metadata":{"executionInfo":{"elapsed":577,"status":"ok","timestamp":1668392182979,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"7LN97U-b2LaD"},"outputs":[],"source":["import pandas as pd\n","from scipy.stats import chi2_contingency\n","\n","dic_result = {\n","    '독립변수':[],\n","    '카이제곱 통계량':[],\n","    'p-value':[],\n","    'df':[],\n","    '기댓값 행렬':[]\n","}\n","df_expected = pd.DataFrame()\n","\n","for i in X.columns:\n","  dic_result['독립변수'].append(i)\n","  A = np.array(X[i])\n","  B = np.array(y['N_category'])\n","\n","  #crosstab 객체의 margins=False여야 \n","  crosstab=pd.crosstab(A, B, margins=False)\n","  # 교차분석 결과값들 일괄계산\n","  # correction=True를 적용하면 Yates' correction이 적용 되어 검정통계량이 보수적으로 더 낮게 나옴\n","  result=chi2_contingency(observed=crosstab, correction=False)\n","  dic_result['카이제곱 통계량'].append(result[0])\n","  dic_result['p-value'].append(result[1])\n","  dic_result['df'].append(result[2])\n","  dic_result['기댓값 행렬'].append(result[3])\n","  df= pd.DataFrame(result[3]).rename(index={0:i}, columns={0:'림프절전이X', 1:'림프절전이O'})\n","  df_expected= pd.concat([df_expected,df])\n","\n","df_result = pd.DataFrame(dic_result)  "]},{"cell_type":"code","execution_count":9,"id":"R8xf-rnzI1MF","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":1000},"executionInfo":{"elapsed":725,"status":"ok","timestamp":1668392184022,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"R8xf-rnzI1MF","outputId":"c4654d62-9fd3-4c3b-ff62-44095e16afe0"},"outputs":[{"data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>독립변수</th>\n","      <th>카이제곱 통계량</th>\n","      <th>p-value</th>\n","      <th>df</th>\n","      <th>기댓값 행렬</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>dx</td>\n","      <td>10.369261</td>\n","      <td>1.567452e-02</td>\n","      <td>3</td>\n","      <td>[[416.502, 440.498], [31.59, 33.41], [4.86, 5....</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>location of cancers</td>\n","      <td>0.910054</td>\n","      <td>6.344309e-01</td>\n","      <td>2</td>\n","      <td>[[229.392, 242.608], [240.57, 254.43], [16.038...</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>number of cancers</td>\n","      <td>72.307456</td>\n","      <td>1.841503e-17</td>\n","      <td>1</td>\n","      <td>[[421.362, 445.638], [64.638, 68.362]]</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>NG</td>\n","      <td>170.987072</td>\n","      <td>7.423847e-38</td>\n","      <td>2</td>\n","      <td>[[95.256, 100.744], [262.926, 278.074], [127.8...</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>HG</td>\n","      <td>55.158552</td>\n","      <td>1.053107e-12</td>\n","      <td>2</td>\n","      <td>[[138.024, 145.976], [255.636, 270.364], [92.3...</td>\n","    </tr>\n","    <tr>\n","      <th>5</th>\n","      <td>HG_score_1</td>\n","      <td>48.170307</td>\n","      <td>3.466976e-11</td>\n","      <td>2</td>\n","      <td>[[40.824, 43.176], [100.116, 105.884], [345.06...</td>\n","    </tr>\n","    <tr>\n","      <th>6</th>\n","      <td>HG_score_2</td>\n","      <td>97.981359</td>\n","      <td>5.291980e-22</td>\n","      <td>2</td>\n","      <td>[[67.554, 71.446], [287.226, 303.774], [131.22...</td>\n","    </tr>\n","    <tr>\n","      <th>7</th>\n","      <td>HG_score_3</td>\n","      <td>20.783127</td>\n","      <td>1.167775e-04</td>\n","      <td>3</td>\n","      <td>[[353.808, 374.192], [76.302, 80.698], [54.432...</td>\n","    </tr>\n","    <tr>\n","      <th>8</th>\n","      <td>DCIS_or_LCIS</td>\n","      <td>185.077813</td>\n","      <td>6.469393e-41</td>\n","      <td>2</td>\n","      <td>[[291.114, 307.886], [40.338, 42.662], [154.54...</td>\n","    </tr>\n","    <tr>\n","      <th>9</th>\n","      <td>DCIS_or_LCIS_type</td>\n","      <td>28.714687</td>\n","      <td>8.386527e-08</td>\n","      <td>1</td>\n","      <td>[[33.534, 35.466], [452.466, 478.534]]</td>\n","    </tr>\n","    <tr>\n","      <th>10</th>\n","      <td>T_category</td>\n","      <td>125.574647</td>\n","      <td>3.439903e-26</td>\n","      <td>4</td>\n","      <td>[[34.506, 36.494], [293.544, 310.456], [148.23...</td>\n","    </tr>\n","    <tr>\n","      <th>11</th>\n","      <td>ER</td>\n","      <td>0.007334</td>\n","      <td>9.317551e-01</td>\n","      <td>1</td>\n","      <td>[[87.48, 92.52], [398.52, 421.48]]</td>\n","    </tr>\n","    <tr>\n","      <th>12</th>\n","      <td>PR</td>\n","      <td>2.513120</td>\n","      <td>1.129022e-01</td>\n","      <td>1</td>\n","      <td>[[173.988, 184.012], [312.012, 329.988]]</td>\n","    </tr>\n","    <tr>\n","      <th>13</th>\n","      <td>HER2</td>\n","      <td>8.181040</td>\n","      <td>4.233048e-03</td>\n","      <td>1</td>\n","      <td>[[367.416, 388.584], [118.584, 125.416]]</td>\n","    </tr>\n","    <tr>\n","      <th>14</th>\n","      <td>HER2_IHC</td>\n","      <td>2.325276</td>\n","      <td>5.076960e-01</td>\n","      <td>3</td>\n","      <td>[[121.986, 129.014], [152.604, 161.396], [169....</td>\n","    </tr>\n","    <tr>\n","      <th>15</th>\n","      <td>HER2_SISH</td>\n","      <td>9.451716</td>\n","      <td>2.109517e-03</td>\n","      <td>1</td>\n","      <td>[[470.448, 497.552], [15.552, 16.448]]</td>\n","    </tr>\n","    <tr>\n","      <th>16</th>\n","      <td>BRCA_mutation</td>\n","      <td>0.895417</td>\n","      <td>3.440136e-01</td>\n","      <td>1</td>\n","      <td>[[484.056, 511.944], [1.944, 2.056]]</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["                   독립변수    카이제곱 통계량       p-value  df  \\\n","0                    dx   10.369261  1.567452e-02   3   \n","1   location of cancers    0.910054  6.344309e-01   2   \n","2     number of cancers   72.307456  1.841503e-17   1   \n","3                    NG  170.987072  7.423847e-38   2   \n","4                    HG   55.158552  1.053107e-12   2   \n","5            HG_score_1   48.170307  3.466976e-11   2   \n","6            HG_score_2   97.981359  5.291980e-22   2   \n","7            HG_score_3   20.783127  1.167775e-04   3   \n","8          DCIS_or_LCIS  185.077813  6.469393e-41   2   \n","9     DCIS_or_LCIS_type   28.714687  8.386527e-08   1   \n","10           T_category  125.574647  3.439903e-26   4   \n","11                   ER    0.007334  9.317551e-01   1   \n","12                   PR    2.513120  1.129022e-01   1   \n","13                 HER2    8.181040  4.233048e-03   1   \n","14             HER2_IHC    2.325276  5.076960e-01   3   \n","15            HER2_SISH    9.451716  2.109517e-03   1   \n","16        BRCA_mutation    0.895417  3.440136e-01   1   \n","\n","                                               기댓값 행렬  \n","0   [[416.502, 440.498], [31.59, 33.41], [4.86, 5....  \n","1   [[229.392, 242.608], [240.57, 254.43], [16.038...  \n","2              [[421.362, 445.638], [64.638, 68.362]]  \n","3   [[95.256, 100.744], [262.926, 278.074], [127.8...  \n","4   [[138.024, 145.976], [255.636, 270.364], [92.3...  \n","5   [[40.824, 43.176], [100.116, 105.884], [345.06...  \n","6   [[67.554, 71.446], [287.226, 303.774], [131.22...  \n","7   [[353.808, 374.192], [76.302, 80.698], [54.432...  \n","8   [[291.114, 307.886], [40.338, 42.662], [154.54...  \n","9              [[33.534, 35.466], [452.466, 478.534]]  \n","10  [[34.506, 36.494], [293.544, 310.456], [148.23...  \n","11                 [[87.48, 92.52], [398.52, 421.48]]  \n","12           [[173.988, 184.012], [312.012, 329.988]]  \n","13           [[367.416, 388.584], [118.584, 125.416]]  \n","14  [[121.986, 129.014], [152.604, 161.396], [169....  \n","15             [[470.448, 497.552], [15.552, 16.448]]  \n","16               [[484.056, 511.944], [1.944, 2.056]]  "]},"execution_count":9,"metadata":{},"output_type":"execute_result"}],"source":["df_result"]},{"cell_type":"code","execution_count":10,"id":"gEBkYOxxNAFH","metadata":{"executionInfo":{"elapsed":5,"status":"ok","timestamp":1668392184334,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"gEBkYOxxNAFH"},"outputs":[],"source":["df_result.to_csv(r'C:\\A\\breastCancer\\ml/output/coll/df_result_cat_corrleation_3.csv')"]},{"cell_type":"code","execution_count":11,"id":"yJ0um8h2JTbF","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":424},"executionInfo":{"elapsed":4,"status":"ok","timestamp":1668392184334,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"yJ0um8h2JTbF","outputId":"38112a9c-52d4-4223-fda5-26ff708bb888"},"outputs":[{"data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>림프절전이X</th>\n","      <th>림프절전이O</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>dx</th>\n","      <td>416.502</td>\n","      <td>440.498</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>31.590</td>\n","      <td>33.410</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>4.860</td>\n","      <td>5.140</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>33.048</td>\n","      <td>34.952</td>\n","    </tr>\n","    <tr>\n","      <th>location of cancers</th>\n","      <td>229.392</td>\n","      <td>242.608</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>240.570</td>\n","      <td>254.430</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>16.038</td>\n","      <td>16.962</td>\n","    </tr>\n","    <tr>\n","      <th>number of cancers</th>\n","      <td>421.362</td>\n","      <td>445.638</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>64.638</td>\n","      <td>68.362</td>\n","    </tr>\n","    <tr>\n","      <th>NG</th>\n","      <td>95.256</td>\n","      <td>100.744</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>262.926</td>\n","      <td>278.074</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>127.818</td>\n","      <td>135.182</td>\n","    </tr>\n","    <tr>\n","      <th>HG</th>\n","      <td>138.024</td>\n","      <td>145.976</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>255.636</td>\n","      <td>270.364</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>92.340</td>\n","      <td>97.660</td>\n","    </tr>\n","    <tr>\n","      <th>HG_score_1</th>\n","      <td>40.824</td>\n","      <td>43.176</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>100.116</td>\n","      <td>105.884</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>345.060</td>\n","      <td>364.940</td>\n","    </tr>\n","    <tr>\n","      <th>HG_score_2</th>\n","      <td>67.554</td>\n","      <td>71.446</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>287.226</td>\n","      <td>303.774</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>131.220</td>\n","      <td>138.780</td>\n","    </tr>\n","    <tr>\n","      <th>HG_score_3</th>\n","      <td>353.808</td>\n","      <td>374.192</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>76.302</td>\n","      <td>80.698</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>54.432</td>\n","      <td>57.568</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>1.458</td>\n","      <td>1.542</td>\n","    </tr>\n","    <tr>\n","      <th>DCIS_or_LCIS</th>\n","      <td>291.114</td>\n","      <td>307.886</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>40.338</td>\n","      <td>42.662</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>154.548</td>\n","      <td>163.452</td>\n","    </tr>\n","    <tr>\n","      <th>DCIS_or_LCIS_type</th>\n","      <td>33.534</td>\n","      <td>35.466</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>452.466</td>\n","      <td>478.534</td>\n","    </tr>\n","    <tr>\n","      <th>T_category</th>\n","      <td>34.506</td>\n","      <td>36.494</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>293.544</td>\n","      <td>310.456</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>148.230</td>\n","      <td>156.770</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>7.776</td>\n","      <td>8.224</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>1.944</td>\n","      <td>2.056</td>\n","    </tr>\n","    <tr>\n","      <th>ER</th>\n","      <td>87.480</td>\n","      <td>92.520</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>398.520</td>\n","      <td>421.480</td>\n","    </tr>\n","    <tr>\n","      <th>PR</th>\n","      <td>173.988</td>\n","      <td>184.012</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>312.012</td>\n","      <td>329.988</td>\n","    </tr>\n","    <tr>\n","      <th>HER2</th>\n","      <td>367.416</td>\n","      <td>388.584</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>118.584</td>\n","      <td>125.416</td>\n","    </tr>\n","    <tr>\n","      <th>HER2_IHC</th>\n","      <td>121.986</td>\n","      <td>129.014</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>152.604</td>\n","      <td>161.396</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>169.128</td>\n","      <td>178.872</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>42.282</td>\n","      <td>44.718</td>\n","    </tr>\n","    <tr>\n","      <th>HER2_SISH</th>\n","      <td>470.448</td>\n","      <td>497.552</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>15.552</td>\n","      <td>16.448</td>\n","    </tr>\n","    <tr>\n","      <th>BRCA_mutation</th>\n","      <td>484.056</td>\n","      <td>511.944</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>1.944</td>\n","      <td>2.056</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["                      림프절전이X   림프절전이O\n","dx                   416.502  440.498\n","1                     31.590   33.410\n","2                      4.860    5.140\n","3                     33.048   34.952\n","location of cancers  229.392  242.608\n","1                    240.570  254.430\n","2                     16.038   16.962\n","number of cancers    421.362  445.638\n","1                     64.638   68.362\n","NG                    95.256  100.744\n","1                    262.926  278.074\n","2                    127.818  135.182\n","HG                   138.024  145.976\n","1                    255.636  270.364\n","2                     92.340   97.660\n","HG_score_1            40.824   43.176\n","1                    100.116  105.884\n","2                    345.060  364.940\n","HG_score_2            67.554   71.446\n","1                    287.226  303.774\n","2                    131.220  138.780\n","HG_score_3           353.808  374.192\n","1                     76.302   80.698\n","2                     54.432   57.568\n","3                      1.458    1.542\n","DCIS_or_LCIS         291.114  307.886\n","1                     40.338   42.662\n","2                    154.548  163.452\n","DCIS_or_LCIS_type     33.534   35.466\n","1                    452.466  478.534\n","T_category            34.506   36.494\n","1                    293.544  310.456\n","2                    148.230  156.770\n","3                      7.776    8.224\n","4                      1.944    2.056\n","ER                    87.480   92.520\n","1                    398.520  421.480\n","PR                   173.988  184.012\n","1                    312.012  329.988\n","HER2                 367.416  388.584\n","1                    118.584  125.416\n","HER2_IHC             121.986  129.014\n","1                    152.604  161.396\n","2                    169.128  178.872\n","3                     42.282   44.718\n","HER2_SISH            470.448  497.552\n","1                     15.552   16.448\n","BRCA_mutation        484.056  511.944\n","1                      1.944    2.056"]},"execution_count":11,"metadata":{},"output_type":"execute_result"}],"source":["df_expected"]},{"cell_type":"code","execution_count":12,"id":"S9n6KdEx32Wb","metadata":{"executionInfo":{"elapsed":2,"status":"ok","timestamp":1668392186396,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"S9n6KdEx32Wb"},"outputs":[],"source":["df_expected.to_csv(r'C:\\A\\breastCancer\\ml\\output\\coll/df_expected_cat_corrleation_3.csv')"]},{"cell_type":"markdown","id":"5p4JnYop2LUQ","metadata":{"id":"5p4JnYop2LUQ"},"source":["# num_col 상관관계분석"]},{"cell_type":"code","execution_count":13,"id":"95jXPt04RcY3","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":292,"status":"ok","timestamp":1668392143315,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"95jXPt04RcY3","outputId":"ff8a6cc8-5bb1-4e47-81dc-3011e3590548"},"outputs":[{"name":"stdout","output_type":"stream","text":["The autoreload extension is already loaded. To reload it, use:\n","  %reload_ext autoreload\n"]}],"source":["import os\n","try:\n","  import google.colab\n","  IN_COLAB = True\n","except:\n","  IN_COLAB = False\n","if not IN_COLAB:\n","    os.chdir(\"..\")\n","from sklearn.datasets import fetch_covtype\n","import random\n","import numpy as np\n","import pandas as pd\n","import lightgbm as lgb\n","from sklearn.ensemble import RandomForestClassifier\n","from sklearn.metrics import accuracy_score, f1_score\n","from sklearn.model_selection import train_test_split\n","%load_ext autoreload\n","%autoreload 2\n"]},{"cell_type":"code","execution_count":14,"id":"u74tPgQKRcRB","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":2837,"status":"ok","timestamp":1668392147697,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"u74tPgQKRcRB","outputId":"9620b0c3-bc7b-468c-f6d7-b0552a98c264"},"outputs":[],"source":["import numpy as np # linear algebra\n","import pandas as pd # data processing, CSV file I/O (e.g. pd.read_csv)\n","import matplotlib.pyplot as plt # data visualization\n","import seaborn as sns # statistical data visualization\n","import os\n","import warnings\n","warnings.filterwarnings(\"ignore\")\n"]},{"cell_type":"code","execution_count":15,"id":"AVv-k9erRcPB","metadata":{"id":"AVv-k9erRcPB"},"outputs":[],"source":["import pandas as pd\n","import numpy as np\n","\n","train_df = pd.read_csv(r'C:\\A\\breastCancer\\ml/train.csv')\n","test_df = pd.read_csv(r\"C:\\A\\breastCancer\\ml/test.csv\")\n"]},{"cell_type":"code","execution_count":16,"id":"2_fzSbUARcMn","metadata":{"id":"2_fzSbUARcMn"},"outputs":[],"source":["# CAT_COL fillna\n","train_df['NG']= train_df['NG'].fillna(2)\n","train_df['HG']= train_df['HG'].fillna(2)\n","train_df['HG_score_1']= train_df['HG_score_1'].fillna(3)\n","train_df['HG_score_2']= train_df['HG_score_2'].fillna(2)\n","train_df['HG_score_3']= train_df['HG_score_3'].fillna(1)\n","train_df['DCIS_or_LCIS_type']= train_df['DCIS_or_LCIS_type'].fillna(2)\n","train_df['T_category']= train_df['T_category'].fillna(1)\n","train_df['ER']= train_df['ER'].fillna(1)\n","train_df['PR']= train_df['PR'].fillna(1)\n","train_df['HER2']= train_df['HER2'].fillna(0)\n","train_df['HER2_IHC']= train_df['HER2_IHC'].fillna(2)\n","train_df['HER2_SISH']= train_df['HER2_SISH'].fillna(0)\n","train_df['BRCA_mutation']= train_df['BRCA_mutation'].fillna(0)\n","\n","# NUL_COL fillna\n","# median\n","train_df['size of cancers'] = train_df['size of cancers'].fillna(train_df['size of cancers'].median())\n","train_df['ER_Allred_score'] = train_df['ER_Allred_score'].fillna(train_df['ER_Allred_score'].median())\n","train_df['PR_Allred_score'] = train_df['PR_Allred_score'].fillna(train_df['PR_Allred_score'].median())\n","train_df['KI-67_LI_percent'] = train_df['KI-67_LI_percent'].fillna(train_df['KI-67_LI_percent'].median())\n","train_df['HER2_SISH_ratio'] = train_df['HER2_SISH_ratio'].fillna(train_df['HER2_SISH_ratio'].median())\n","\n","\n","target_name = ['N_category']\n","\n","# 명목형 변수\n","cat_col_names = [\n","    'dx','location of cancers', 'number of cancers', 'NG','HG','HG_score_1','HG_score_2','HG_score_3','DCIS_or_LCIS','DCIS_or_LCIS_type','T_category','ER','PR','HER2','HER2_IHC','HER2_SISH',\"BRCA_mutation\"\n","]\n","\n","for col_name in cat_col_names:\n","    cat_values = np.unique(train_df[col_name])\n","    \n","    #print(f\"{col_name}: {cat_values}\\n\")\n","\n","\n","# 수치형 변수\n","num_col_names = [\n","    'age','size of cancers','ER_Allred_score','PR_Allred_score','KI-67_LI_percent','HER2_SISH_ratio'\n","]\n","\n","feature_columns = (\n","    num_col_names + cat_col_names + target_name)\n","\n","train_df = train_df.loc[:,feature_columns]\n","\n","\n","test = test_df.loc[:,num_col_names + cat_col_names]\n","\n","\n","X = train_df.loc[:,num_col_names]\n","y = train_df.loc[:,target_name]\n"]},{"cell_type":"code","execution_count":17,"id":"0OA0EOHjW_un","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":9,"status":"ok","timestamp":1668064335858,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"0OA0EOHjW_un","outputId":"be6878b0-5a91-4e32-d8db-5a6e1cee2655"},"outputs":[{"name":"stdout","output_type":"stream","text":["<class 'pandas.core.frame.DataFrame'>\n","RangeIndex: 1000 entries, 0 to 999\n","Data columns (total 6 columns):\n"," #   Column            Non-Null Count  Dtype  \n","---  ------            --------------  -----  \n"," 0   age               1000 non-null   int64  \n"," 1   size of cancers   1000 non-null   float64\n"," 2   ER_Allred_score   1000 non-null   float64\n"," 3   PR_Allred_score   1000 non-null   float64\n"," 4   KI-67_LI_percent  1000 non-null   float64\n"," 5   HER2_SISH_ratio   1000 non-null   float64\n","dtypes: float64(5), int64(1)\n","memory usage: 47.0 KB\n"]}],"source":["X.info()"]},{"cell_type":"code","execution_count":18,"id":"8gEYpXGsXBic","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":7,"status":"ok","timestamp":1668064335858,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"8gEYpXGsXBic","outputId":"b50021a5-2780-4e6d-980a-b17397405bc0"},"outputs":[{"name":"stdout","output_type":"stream","text":["<class 'pandas.core.frame.DataFrame'>\n","RangeIndex: 1000 entries, 0 to 999\n","Data columns (total 1 columns):\n"," #   Column      Non-Null Count  Dtype\n","---  ------      --------------  -----\n"," 0   N_category  1000 non-null   int64\n","dtypes: int64(1)\n","memory usage: 7.9 KB\n"]}],"source":["y.info()"]},{"cell_type":"code","execution_count":19,"id":"Icj65wtQXvmU","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":6,"status":"ok","timestamp":1668064786075,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"Icj65wtQXvmU","outputId":"ebda3f1f-3b08-401a-95f7-8b5540f929fa"},"outputs":[],"source":["import pandas as pd\n","import scipy.stats as stats\n","\n","dic_result = {\n","    '독립변수':[],\n","    'correlation':[],\n","    'p-value':[]\n","}\n","df_expected = pd.DataFrame()\n","\n","for i in X.columns:\n","  dic_result['독립변수'].append(i)\n","  A = (X[i])\n","  B = y['N_category']\n","  result = stats.pearsonr(A,B)\n","  dic_result['correlation'].append(result[0])\n","  dic_result['p-value'].append(result[1])\n","\n","df_result = pd.DataFrame(dic_result)  "]},{"cell_type":"code","execution_count":20,"id":"EJ_BvYFcZ908","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":238},"executionInfo":{"elapsed":14,"status":"ok","timestamp":1668064789747,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"EJ_BvYFcZ908","outputId":"7d5e0cb9-5a86-4ccb-bb88-186e38152d3b"},"outputs":[{"data":{"text/html":["<div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>독립변수</th>\n","      <th>correlation</th>\n","      <th>p-value</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>age</td>\n","      <td>0.069017</td>\n","      <td>2.908133e-02</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>size of cancers</td>\n","      <td>0.287994</td>\n","      <td>1.494569e-20</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>ER_Allred_score</td>\n","      <td>0.054848</td>\n","      <td>8.299414e-02</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>PR_Allred_score</td>\n","      <td>-0.047400</td>\n","      <td>1.341620e-01</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>KI-67_LI_percent</td>\n","      <td>0.188617</td>\n","      <td>1.841873e-09</td>\n","    </tr>\n","    <tr>\n","      <th>5</th>\n","      <td>HER2_SISH_ratio</td>\n","      <td>0.034231</td>\n","      <td>2.795056e-01</td>\n","    </tr>\n","  </tbody>\n","</table>\n","</div>"],"text/plain":["               독립변수  correlation       p-value\n","0               age     0.069017  2.908133e-02\n","1   size of cancers     0.287994  1.494569e-20\n","2   ER_Allred_score     0.054848  8.299414e-02\n","3   PR_Allred_score    -0.047400  1.341620e-01\n","4  KI-67_LI_percent     0.188617  1.841873e-09\n","5   HER2_SISH_ratio     0.034231  2.795056e-01"]},"execution_count":20,"metadata":{},"output_type":"execute_result"}],"source":["df_result"]},{"cell_type":"code","execution_count":null,"id":"40666f6d","metadata":{},"outputs":[],"source":[]},{"cell_type":"code","execution_count":21,"id":"J-Rpy9R1Z9uc","metadata":{"id":"J-Rpy9R1Z9uc"},"outputs":[],"source":["df_result.to_csv(r'C:\\A\\breastCancer\\ml\\output\\coll/df_result_num_corrleation_3.csv')"]},{"cell_type":"markdown","id":"jsZ4hxo0RasR","metadata":{"id":"jsZ4hxo0RasR"},"source":["# 분류 시작"]},{"cell_type":"code","execution_count":null,"id":"N5sYFOBO3G9r","metadata":{"id":"N5sYFOBO3G9r"},"outputs":[],"source":["SEED = 88\n","NSPLIT = 10\n","\n","def seed_everything(seed):\n","    random.seed(seed)\n","    os.environ['PYTHONHASHSEED'] = str(seed)\n","    np.random.seed(seed)\n","\n","seed_everything(SEED) # Seed 고정"]},{"cell_type":"code","execution_count":null,"id":"3OsXXqSL4qJy","metadata":{"id":"3OsXXqSL4qJy"},"outputs":[],"source":["\"\"\"\n","# split the dataset into the training set and test set\n","from sklearn.model_selection import train_test_split\n","X_train, X_test, y_train, y_test = train_test_split(X, y, test_size = 0.2, random_state = SEED)\n","\"\"\"\n","\n","from sklearn.model_selection import KFold\n","\n","X = np.array(X)                 #np.array(df.iloc[:, :-1]) # class 열 제외한 feature 열들 모음 -> array 변환\n","y = np.array(y)                           #df['class']\n","\n"]},{"cell_type":"code","execution_count":null,"id":"CFNUvKvYe6oy","metadata":{"id":"CFNUvKvYe6oy"},"outputs":[],"source":["# split 개수, 셔플 여부 및 seed 설정\n","kf = KFold(n_splits = NSPLIT, shuffle = True, random_state = SEED)\n","\n","# split 개수 스텝 만큼 train, test 데이터셋을 매번 분할\n","for train_index, test_index in kf.split(X):\n","    X_train, X_test = X[train_index], X[test_index]\n","    y_train, y_test = y[train_index], y[test_index]"]},{"cell_type":"markdown","id":"m_nZ4Q0kK-88","metadata":{"id":"m_nZ4Q0kK-88"},"source":["##lightGBM"]},{"cell_type":"code","execution_count":null,"id":"b7IvlN1ife9E","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":8,"status":"ok","timestamp":1668049876052,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"b7IvlN1ife9E","outputId":"181fa5da-ca9c-4d66-ac1b-6c1f3e39ca26"},"outputs":[{"name":"stdout","output_type":"stream","text":["accuracy 각 분할의 정확도 기록 : [0.85, 0.79, 0.73, 0.77, 0.75, 0.71, 0.77, 0.77, 0.75, 0.79]\n","accuracy 평균 정확도 : 0.768\n","f1_score 각 분할의 정확도 기록 : [0.8421052631578948, 0.8073394495412846, 0.7157894736842104, 0.7722772277227723, 0.7252747252747253, 0.7184466019417476, 0.8067226890756303, 0.7766990291262137, 0.7368421052631577, 0.817391304347826]\n","f1_score 평균 정확도 : 0.7718887869135462\n"]}],"source":["import lightgbm as lgb\n","\n","from sklearn.metrics import accuracy_score\n","from sklearn.metrics import f1_score\n","\n","accuracy_history = []\n","f1_score_history = []\n","\n","# K-fold 검증 과정으로 실제 랜덤 포레스트 모델을 학습하여 정확도 평균을 내는 방법\n","for train_index, test_index in kf.split(X):\n","    X_train, X_test = X[train_index], X[test_index]\n","    y_train, y_test = y[train_index], y[test_index]\n","\n","    \n","    clf = lgb.LGBMClassifier(n_estimators=5, random_state=SEED) # 모델 선언\n","    clf.fit(X_train,y_train) # 모델 학습\n","    y_pred=clf.predict(X_test) # 예측 라벨\n","    accuracy_history.append(accuracy_score(y_pred, y_test)) # 정확도 측정 및 기록\n","    f1_score_history.append(f1_score(y_pred, y_test)) # 정확도 측정 및 기록\n","\n","print(\"accuracy 각 분할의 정확도 기록 :\", accuracy_history)\n","print(\"accuracy 평균 정확도 :\", np.mean(accuracy_history))\n","\n","print(\"f1_score 각 분할의 정확도 기록 :\", f1_score_history)\n","print(\"f1_score 평균 정확도 :\", np.mean(f1_score_history))"]},{"cell_type":"markdown","id":"jWRYQAZvLIE4","metadata":{"id":"jWRYQAZvLIE4"},"source":["## catboost classififier"]},{"cell_type":"code","execution_count":null,"id":"2LTsm2quiJpn","metadata":{"id":"2LTsm2quiJpn"},"outputs":[],"source":["!pip install catboost"]},{"cell_type":"code","execution_count":null,"id":"wxpSPSLxhnlL","metadata":{"id":"wxpSPSLxhnlL"},"outputs":[],"source":["from catboost import CatBoostClassifier\n","\n","from sklearn.metrics import accuracy_score\n","from sklearn.metrics import f1_score\n","\n","accuracy_history = []\n","f1_score_history = []\n","\n","# K-fold 검증 과정으로 실제 랜덤 포레스트 모델을 학습하여 정확도 평균을 내는 방법\n","for train_index, test_index in kf.split(X):\n","    X_train, X_test = X[train_index], X[test_index]\n","    y_train, y_test = y[train_index], y[test_index]\n","\n","    clf = CatBoostClassifier(\n","      iterations=5, \n","      learning_rate=0.1, \n","      loss_function='CrossEntropy',\n","      random_state=SEED) # 모델 선언\n","\n","    clf.fit(X_train,y_train) # 모델 학습\n","    y_pred=clf.predict(X_test) # 예측 라벨\n","    accuracy_history.append(accuracy_score(y_pred, y_test)) # 정확도 측정 및 기록\n","    f1_score_history.append(f1_score(y_pred, y_test)) # 정확도 측정 및 기록\n","\n","print(\"accuracy 각 분할의 정확도 기록 :\", accuracy_history)\n","print(\"accuracy 평균 정확도 :\", np.mean(accuracy_history))\n","\n","print(\"f1_score 각 분할의 정확도 기록 :\", f1_score_history)\n","print(\"f1_score 평균 정확도 :\", np.mean(f1_score_history))"]},{"cell_type":"markdown","id":"y5-MxpGrLQqz","metadata":{"id":"y5-MxpGrLQqz"},"source":["## test preidiction"]},{"cell_type":"code","execution_count":null,"id":"MnioXI8z7BMk","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":7,"status":"ok","timestamp":1668045538873,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"MnioXI8z7BMk","outputId":"d5692318-131d-4949-9ea2-c491818feebf"},"outputs":[{"data":{"text/plain":["(250, pandas.core.series.Series)"]},"execution_count":42,"metadata":{},"output_type":"execute_result"}],"source":["test_df_id = test_df.loc[:,'ID']\n","len(test_df_id),type(test_df_id)"]},{"cell_type":"code","execution_count":null,"id":"omc7kGz4Ddht","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":6,"status":"ok","timestamp":1668045538873,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"omc7kGz4Ddht","outputId":"bb5b26c4-f5ac-4bad-82dd-5cdd95e3c2f2"},"outputs":[{"data":{"text/plain":["(250, numpy.ndarray)"]},"execution_count":43,"metadata":{},"output_type":"execute_result"}],"source":["array = np.array(test_df_id.values)\n","len(array),type(array)"]},{"cell_type":"code","execution_count":null,"id":"o7lEnDev6ms6","metadata":{"colab":{"base_uri":"https://localhost:8080/"},"executionInfo":{"elapsed":4,"status":"ok","timestamp":1668045538873,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"o7lEnDev6ms6","outputId":"629e46c1-6933-4329-9fce-2f79d1738ee7"},"outputs":[{"data":{"text/plain":["(250, numpy.ndarray)"]},"execution_count":44,"metadata":{},"output_type":"execute_result"}],"source":["test_pred = clf.predict(test)\n","len(test_pred),type(test_pred)"]},{"cell_type":"code","execution_count":null,"id":"FQ_yazQZDb6T","metadata":{"id":"FQ_yazQZDb6T"},"outputs":[],"source":["dic_pred={\n","    'ID' : array,\n","    'N_category':test_pred\n","}"]},{"cell_type":"code","execution_count":null,"id":"lvET5Og1Bvrl","metadata":{"id":"lvET5Og1Bvrl"},"outputs":[],"source":["df_test_pred = pd.DataFrame(dic_pred)"]},{"cell_type":"code","execution_count":null,"id":"GroRoyhLBvpU","metadata":{"colab":{"base_uri":"https://localhost:8080/","height":424},"executionInfo":{"elapsed":5,"status":"ok","timestamp":1668045540043,"user":{"displayName":"박혜정","userId":"13305762505933234909"},"user_tz":-540},"id":"GroRoyhLBvpU","outputId":"266af14e-72f4-4fc9-8b04-467d054ac2fa"},"outputs":[{"data":{"text/html":["\n","  <div id=\"df-8e1de0da-1e31-4e34-9365-11768e9b475d\">\n","    <div class=\"colab-df-container\">\n","      <div>\n","<style scoped>\n","    .dataframe tbody tr th:only-of-type {\n","        vertical-align: middle;\n","    }\n","\n","    .dataframe tbody tr th {\n","        vertical-align: top;\n","    }\n","\n","    .dataframe thead th {\n","        text-align: right;\n","    }\n","</style>\n","<table border=\"1\" class=\"dataframe\">\n","  <thead>\n","    <tr style=\"text-align: right;\">\n","      <th></th>\n","      <th>ID</th>\n","      <th>N_category</th>\n","    </tr>\n","  </thead>\n","  <tbody>\n","    <tr>\n","      <th>0</th>\n","      <td>BC_01_0011</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>1</th>\n","      <td>BC_01_0220</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>2</th>\n","      <td>BC_01_0233</td>\n","      <td>0</td>\n","    </tr>\n","    <tr>\n","      <th>3</th>\n","      <td>BC_01_0258</td>\n","      <td>0</td>\n","    </tr>\n","    <tr>\n","      <th>4</th>\n","      <td>BC_01_0260</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>...</th>\n","      <td>...</td>\n","      <td>...</td>\n","    </tr>\n","    <tr>\n","      <th>245</th>\n","      <td>BC_01_3328</td>\n","      <td>1</td>\n","    </tr>\n","    <tr>\n","      <th>246</th>\n","      <td>BC_01_3404</td>\n","      <td>0</td>\n","    </tr>\n","    <tr>\n","      <th>247</th>\n","      <td>BC_01_3418</td>\n","      <td>0</td>\n","    </tr>\n","    <tr>\n","      <th>248</th>\n","      <td>BC_01_3438</td>\n","      <td>0</td>\n","    </tr>\n","    <tr>\n","      <th>249</th>\n","      <td>BC_01_3446</td>\n","      <td>0</td>\n","    </tr>\n","  </tbody>\n","</table>\n","<p>250 rows × 2 columns</p>\n","</div>\n","      <button class=\"colab-df-convert\" onclick=\"convertToInteractive('df-8e1de0da-1e31-4e34-9365-11768e9b475d')\"\n","              title=\"Convert this dataframe to an interactive table.\"\n","              style=\"display:none;\">\n","        \n","  <svg xmlns=\"http://www.w3.org/2000/svg\" height=\"24px\"viewBox=\"0 0 24 24\"\n","       width=\"24px\">\n","    <path d=\"M0 0h24v24H0V0z\" fill=\"none\"/>\n","    <path d=\"M18.56 5.44l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94zm-11 1L8.5 8.5l.94-2.06 2.06-.94-2.06-.94L8.5 2.5l-.94 2.06-2.06.94zm10 10l.94 2.06.94-2.06 2.06-.94-2.06-.94-.94-2.06-.94 2.06-2.06.94z\"/><path d=\"M17.41 7.96l-1.37-1.37c-.4-.4-.92-.59-1.43-.59-.52 0-1.04.2-1.43.59L10.3 9.45l-7.72 7.72c-.78.78-.78 2.05 0 2.83L4 21.41c.39.39.9.59 1.41.59.51 0 1.02-.2 1.41-.59l7.78-7.78 2.81-2.81c.8-.78.8-2.07 0-2.86zM5.41 20L4 18.59l7.72-7.72 1.47 1.35L5.41 20z\"/>\n","  </svg>\n","      </button>\n","      \n","  <style>\n","    .colab-df-container {\n","      display:flex;\n","      flex-wrap:wrap;\n","      gap: 12px;\n","    }\n","\n","    .colab-df-convert {\n","      background-color: #E8F0FE;\n","      border: none;\n","      border-radius: 50%;\n","      cursor: pointer;\n","      display: none;\n","      fill: #1967D2;\n","      height: 32px;\n","      padding: 0 0 0 0;\n","      width: 32px;\n","    }\n","\n","    .colab-df-convert:hover {\n","      background-color: #E2EBFA;\n","      box-shadow: 0px 1px 2px rgba(60, 64, 67, 0.3), 0px 1px 3px 1px rgba(60, 64, 67, 0.15);\n","      fill: #174EA6;\n","    }\n","\n","    [theme=dark] .colab-df-convert {\n","      background-color: #3B4455;\n","      fill: #D2E3FC;\n","    }\n","\n","    [theme=dark] .colab-df-convert:hover {\n","      background-color: #434B5C;\n","      box-shadow: 0px 1px 3px 1px rgba(0, 0, 0, 0.15);\n","      filter: drop-shadow(0px 1px 2px rgba(0, 0, 0, 0.3));\n","      fill: #FFFFFF;\n","    }\n","  </style>\n","\n","      <script>\n","        const buttonEl =\n","          document.querySelector('#df-8e1de0da-1e31-4e34-9365-11768e9b475d button.colab-df-convert');\n","        buttonEl.style.display =\n","          google.colab.kernel.accessAllowed ? 'block' : 'none';\n","\n","        async function convertToInteractive(key) {\n","          const element = document.querySelector('#df-8e1de0da-1e31-4e34-9365-11768e9b475d');\n","          const dataTable =\n","            await google.colab.kernel.invokeFunction('convertToInteractive',\n","                                                     [key], {});\n","          if (!dataTable) return;\n","\n","          const docLinkHtml = 'Like what you see? Visit the ' +\n","            '<a target=\"_blank\" href=https://colab.research.google.com/notebooks/data_table.ipynb>data table notebook</a>'\n","            + ' to learn more about interactive tables.';\n","          element.innerHTML = '';\n","          dataTable['output_type'] = 'display_data';\n","          await google.colab.output.renderOutput(dataTable, element);\n","          const docLink = document.createElement('div');\n","          docLink.innerHTML = docLinkHtml;\n","          element.appendChild(docLink);\n","        }\n","      </script>\n","    </div>\n","  </div>\n","  "],"text/plain":["             ID  N_category\n","0    BC_01_0011           1\n","1    BC_01_0220           1\n","2    BC_01_0233           0\n","3    BC_01_0258           0\n","4    BC_01_0260           1\n","..          ...         ...\n","245  BC_01_3328           1\n","246  BC_01_3404           0\n","247  BC_01_3418           0\n","248  BC_01_3438           0\n","249  BC_01_3446           0\n","\n","[250 rows x 2 columns]"]},"execution_count":47,"metadata":{},"output_type":"execute_result"}],"source":["df_test_pred"]},{"cell_type":"code","execution_count":null,"id":"fDMxIgDrECf-","metadata":{"id":"fDMxIgDrECf-"},"outputs":[],"source":["df_test_pred.to_csv(f'/content/drive/MyDrive/project/BreastCancer/ml/output/lgbm_{SEED}_{f1score}.csv')"]},{"cell_type":"code","execution_count":null,"id":"02CEa4WkJ73Z","metadata":{"id":"02CEa4WkJ73Z"},"outputs":[],"source":[]},{"cell_type":"markdown","id":"iX5k7FYmhkZF","metadata":{"id":"iX5k7FYmhkZF"},"source":[]},{"cell_type":"markdown","id":"x2tn2NVghkXT","metadata":{"id":"x2tn2NVghkXT"},"source":[]},{"cell_type":"markdown","id":"-NGmlZQUhkVX","metadata":{"id":"-NGmlZQUhkVX"},"source":[]},{"cell_type":"markdown","id":"0lTzOxWuhkTQ","metadata":{"id":"0lTzOxWuhkTQ"},"source":[]},{"cell_type":"markdown","id":"prtKpMlbhkRr","metadata":{"id":"prtKpMlbhkRr"},"source":[]},{"cell_type":"markdown","id":"zUfdQ8RuhkPa","metadata":{"id":"zUfdQ8RuhkPa"},"source":[]},{"cell_type":"markdown","id":"yOC0ao3ihkNS","metadata":{"id":"yOC0ao3ihkNS"},"source":[]},{"cell_type":"code","execution_count":null,"id":"PWHQvKZ6hk04","metadata":{"id":"PWHQvKZ6hk04"},"outputs":[],"source":["\"\"\"import lightgbm as lgb\n","clf = lgb.LGBMClassifier()\n","clf\n","clf.fit(X_train,y_train)\n","# predict the results\n","y_pred=clf.predict(X_test)\n","# view accuracy\n","from sklearn.metrics import accuracy_score\n","from sklearn.metrics import f1_score\n","\n","accuracy=accuracy_score(y_pred, y_test)\n","print('LightGBM Model accuracy score: {0:0.4f}'.format(accuracy_score(y_test, y_pred)))\n","y_pred_train = clf.predict(X_train)\n","print('Training-set accuracy score: {0:0.4f}'. format(accuracy_score(y_train, y_pred_train)))\n","\n","\n","# print the scores on training and test set\n","print('Training set score: {:.4f}'.format(clf.score(X_train, y_train)))\n","\n","print('Test set score: {:.4f}'.format(clf.score(X_test, y_test)))\n","from sklearn.metrics import accuracy_score\n","accuracy=accuracy_score(y_pred, y_test)\n","f1score=f1_score(y_pred, y_test)\n","print('LightGBM Model accuracy score: {0:0.4f}'.format(accuracy_score(y_test, y_pred)))\n","print('LightGBM Model f1score score: {0:0.4f}'.format(f1_score(y_pred, y_test)))\"\"\""]},{"cell_type":"code","execution_count":null,"id":"I6o4sZ-oiXHQ","metadata":{"id":"I6o4sZ-oiXHQ"},"outputs":[],"source":["\"\"\"from catboost import CatBoostClassifier\n","clf = CatBoostClassifier(\n","    iterations=5, \n","    learning_rate=0.1, \n","    loss_function='CrossEntropy'\n",")\n","clf.fit(X_train,y_train)\n","# predict the results\n","y_pred=clf.predict(X_test)\n","# view accuracy\n","from sklearn.metrics import accuracy_score\n","accuracy=accuracy_score(y_pred, y_test)\n","print('catboost classififier Model accuracy score: {0:0.4f}'.format(accuracy_score(y_test, y_pred)))\n","y_pred_train = clf.predict(X_train)\n","print('Training-set accuracy score: {0:0.4f}'. format(accuracy_score(y_train, y_pred_train)))\n","# print the scores on training and test set\n","\n","print('Training set score: {:.4f}'.format(clf.score(X_train, y_train)))\n","\n","print('Test set score: {:.4f}'.format(clf.score(X_test, y_test)))\n","from sklearn.metrics import accuracy_score\n","accuracy=accuracy_score(y_pred, y_test)\n","print('catboost classififier Model accuracy score: {0:0.4f}'.format(accuracy_score(y_test, y_pred)))\"\"\""]},{"cell_type":"code","execution_count":null,"id":"Uqq7JqXBiXB-","metadata":{"id":"Uqq7JqXBiXB-"},"outputs":[],"source":[]}],"metadata":{"accelerator":"GPU","colab":{"collapsed_sections":[],"provenance":[]},"gpuClass":"standard","kernelspec":{"display_name":"Python 3.7.13 ('ed4')","language":"python","name":"python3"},"language_info":{"codemirror_mode":{"name":"ipython","version":3},"file_extension":".py","mimetype":"text/x-python","name":"python","nbconvert_exporter":"python","pygments_lexer":"ipython3","version":"3.7.13"},"vscode":{"interpreter":{"hash":"9d933d0ad42eca8b8a8fc5fd44568ab72bdb678767ba8d28729d96933f814f30"}}},"nbformat":4,"nbformat_minor":5}
